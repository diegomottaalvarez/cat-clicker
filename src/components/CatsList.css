ul {
  list-style: none;
  font-size: 1.125em;
  margin-top: 0;
  padding-left: 0;
}

li {
  margin-bottom: 1rem;
  cursor: pointer;
}

.cat-selection-button {
  width: 92%;
  font-size: 1.75rem;
  cursor: pointer;
}

.cat-selection-button:hover {
  transform: rotate(5deg);
  /* transition: transform 0.3s; */
  cursor: pointer;
}

@media (max-width: 900px) {
  ul {
    padding-left: 0;
    width: 100%;
  }

  li {
    margin-bottom: 1.35rem;
  }

  .cat-selection-button {
    width: 100%;
    height: 3rem;
  }
}
