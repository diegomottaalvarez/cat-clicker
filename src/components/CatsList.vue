<template>
  <ul class="names" v-if="catsList.length > 0" data-testid="catsList">
    <li v-for="cat of catsList" :ref="cat.id" class="cat-name"
      ><button
        role="button"
        class="cat-selection-button"
        @click="selectCat(cat)"
        >{{ cat.breeds[0].name }}</button
      ></li
    ></ul
  >
</template>
<script>
export default {
  props: {
    catsList: { type: Array },
  },
  methods: {
    selectCat(cat) {
      window.scrollTo({ top, behavior: 'smooth' });
      this.$emit('catSelection', cat);
    },
  },
};
</script>
<style scoped>
@import './CatsList.css';
</style>
