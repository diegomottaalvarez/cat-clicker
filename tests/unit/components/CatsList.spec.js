import { shallowMount } from '@vue/test-utils';
import CatsList from '@/components/CatsList';
import { catsListMock } from '@/helpers/getCats';

describe('CatsList component', () => {
  window.scrollTo = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(CatsList, {
      propsData: {
        catsList: catsListMock,
      },
    });
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  test('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should paint catslist when data is received', async () => {
    let list = wrapper.find('[data-testid="catsList"]');
    expect(list.exists()).toBeTruthy();
  });

  test('should set selected cat when name is clicked', async () => {
    const name = wrapper.find('.cat-selection-button');

    await wrapper.vm.$nextTick(); // Wait until $emits have been handled

    name.trigger('click');
    expect(wrapper.emitted().catSelection).toBeTruthy();
  });
});
