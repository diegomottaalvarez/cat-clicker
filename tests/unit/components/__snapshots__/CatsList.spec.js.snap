// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatsList component should match the snapshot 1`] = `
<ul class="names" data-testid="catsList">
  <li class="cat-name"><button role="button" class="cat-selection-button">Burmese</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Burmilla</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Devon Rex</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Devon Rex</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Exotic Shorthair</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Exotic Shorthair</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Exotic Shorthair</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Korat</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Manx</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Manx</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Ocicat</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Oriental</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Oriental</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Scottish Fold</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Siamese</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Somali</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">European Burmese</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Somali</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Tonkinese</button></li>
  <li class="cat-name"><button role="button" class="cat-selection-button">Abyssinian</button></li>
</ul>
`;
